designs/ami-selector.md

```diff
@@ -0,0 +1,62 @@
+# Background
+
+The `AMISelector` field of the [`v1alpha1` `AWSNodeTemplate` resource](/pkg/apis/v1alpha1/awsnodetemplate.go)
+is a key-value map with some special key handling (e.g. `aws-ids` is used to pass image IDs).
+
+Most relevant to this design document was the `name` field, which was passed as the `name` filter to
+[`DescribeImages` API](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeImages.html)
+call. This field was removed as a special case in #2978, as there is a risk that without filtering
+by owner, AMI impersonation could happen with various risks (e.g. security vulnerabilities, malware
+such as mining, or cost implications).
+
+While the `Name` tag is still usable, AMI tags are private to an account (so if you want
+to use the AWS bottlerocket AMIs, you'd need to tag them yourself in each account that
+you use)
+
+# Solutions
+
+1. Restore the `name` special case and add a new `owners` special case that is passed
+as the `Owners` argument to `DescribeImages` (this arguments supports both aliases
+such as `self` and `amazon` as well as AWS account IDs). Make `self,amazon` the default
+for this solution.
+
+2. Create a v1alpha2 AMISelector with a much more flexible type than go's `map[string]string` and
+deprecate v1alpha1. The new AMISelector could have an interface much closer to `DescribeImages`.
+
+An example of this might be:
+
+```
+amiSelector:
+  owners:
+    - self
+    - 1234567890
+  name: my-ami
+  imageids:
+    - ami-abcd1234
+    - ami-2345cdef
+  filters:
+    tag:Version: v1.2.3,v1.2.4
+    tag:ThisShouldExist:
+```
+
+I think there's still some discussion for what this should look like, and how closely it should
+map to DescribeImages (here I've taken the approach that the top-level fields all map to an
+argument to DescribeImages)
+
+`owners` would default to `self,amazon`
+
+3. Add an `AMIOwners` list field to v1alpha1 (default `["self","amazon"])` and restore `name` as a special case.
+
+# Recommendations
+
+Solution 1 is pretty much already implemented with #3204 - it's very straightforward, but does
+create a new special case of `owners` that would conflict with any existing use of `owners` as
+an AMI tag.
+
+Solution 3 provides a workaround to the `owners` tag conflict but seems unnecessary and inelegant
+just to provide a way to avoid a theoretical conflict.
+
+Solution 2 is the most work, but removes any of the special cases, and provides a much more flexible
+approach to AMI filtering in general - by allowing filters other than `tag:`, users will gain
+more power.
+
```

Comment:
Writing down my pros and cons here for each solution. Unclear to me which one we should go for, or if theres is a better alternative. Can you come present this to Working Group?

Solution 1
- Pros: Easily solves the problem
- Cons: Can break users that have `owners` and `name` tags in their AWSNodeTemplate already. Adds in multi-case runtime defaulting that may be ambiguous to new users

Solution 2: 
- Pros: Scales to the future a bit better as any EC2 API user can easily put their configs in.
- Cons: Relies heavily on the structure of the EC2 API. Any diff in EC2 APIs in the future will be confusing. This gives less control to Karpenter for the future of how we want this API to look.

Solution 3: 
- Pros: Doesn't break any users
- Cons: Adds config bloat into the AWSNodeTemplate. The field would be misplaced.